deployment/configenv/xml_jlibpt/EnvHelper.cpp

```diff
@@ -0,0 +1,376 @@
+  }
+
+  return bIsValid;
+}
+
+const char* EnvHelper::getXMLTagName(const char* name)
+{
+
+   if (!name)
+      throw MakeStringException(CfgEnvErrorCode::InvalidParams, "Null string in getXMLTagName");
+
+   const char * nameLC = (StringBuffer(name).toLowerCase()).str();
+
+   if (!strcmp(nameLC, "hardware") || !strcmp(nameLC, "hd"))
+      return XML_TAG_HARDWARE;
+   else if (!strcmp(nameLC, "computer") || !strcmp(nameLC, "cmpt"))
+      return XML_TAG_COMPUTER;
+   else if (!strcmp(nameLC, "software") || !strcmp(nameLC, "sw"))
+      return XML_TAG_SOFTWARE;
+   else if (!strcmp(nameLC, "envsettings") || !strcmp(nameLC, "es"))
+      return "EnvSettings";
+   else if (!strcmp(nameLC, "programs") || !strcmp(nameLC, "pg") ||
+            !strcmp(nameLC, "build") || !strcmp(nameLC, "bd"))
+      return "Programs/Build";
+   else if (!strcmp(nameLC, "directories") || !strcmp(nameLC, "dirs"))
+      return XML_TAG_DIRECTORIES;
+   else if (!strcmp(nameLC, "roxie") || !strcmp(nameLC, "roxiecluster"))
+      return XML_TAG_ROXIECLUSTER;
+   else if (!strcmp(nameLC, "thor") || !strcmp(nameLC, "thorcluster"))
+      return XML_TAG_THORCLUSTER;
+   else if (!strcmp(nameLC, "dali") || !strcmp(nameLC, "daliserverprocess") ||
+            !strcmp(nameLC, "dalisrv"))
+      return XML_TAG_DALISERVERPROCESS;
+   else if (!strcmp(nameLC, "dafile") || !strcmp(nameLC, "dafilesrv") ||
+            !strcmp(nameLC, "dafileserverprocess"))
+      return XML_TAG_DAFILESERVERPROCESS;
+   else if (!strcmp(nameLC, "dfu") || !strcmp(nameLC, "dfusrv") ||
+            !strcmp(nameLC, "dfuserverprocess"))
+      return  "DfuServerProcess";
+   else if (!strcmp(nameLC, "dropzone"))
+      return XML_TAG_DROPZONE;
+   else if (!strcmp(nameLC, "eclcc"))
+      return "EclCCProcess";
+   else if (!strcmp(nameLC, "eclccsrv") || !strcmp(nameLC, "eclccserver") || !strcmp(nameLC, "eclcc"))
+      return XML_TAG_ECLCCSERVERPROCESS;
+   else if (!strcmp(nameLC, "esp") || !strcmp(nameLC, "espprocess"))
+      return XML_TAG_ESPPROCESS;
+   else if (!strcmp(nameLC, "espsvc") || !strcmp(nameLC, "espservice") || !strcmp(nameLC, "espsmc") ||
+            !strcmp(nameLC, "ws_sql") || !strcmp(nameLC, "DynamicESDL") || !strcmp(nameLC, "wslogging") ||
+            !strcmp(nameLC, "ws_ecl"))
+      return XML_TAG_ESPSERVICE;
+   else if (!strcmp(nameLC, "binding") || !strcmp(nameLC, "espbinding"))
+      return XML_TAG_ESPBINDING;
+   else if (!strcmp(nameLC, "sasha") || !strcmp(nameLC, "sashasrv"))
+      return XML_TAG_SASHA_SERVER_PROCESS;
+   else if (!strcmp(nameLC, "eclsch") || !strcmp(nameLC, "eclscheduler"))
+      return XML_TAG_ECLSCHEDULERPROCESS;
+   else if (!strcmp(nameLC, "agent") || !strcmp(nameLC, "eclagent"))
+      return XML_TAG_ECLAGENTPROCESS;
+   else if (!strcmp(nameLC, "topology") || !strcmp(nameLC, "topo"))
+      return XML_TAG_TOPOLOGY;
+   else if (!strcmp(nameLC, "ftslave") || !strcmp(nameLC, "ftslaveprocess"))
+      return "FTSlaveProcess";
+
```

Comment:
Another option to get the xml tag could be to use a std::map<std::string, std::string>. Always insert the key value as a lowercase string. This would be much faster lookup if speed is an issue.

Reply:
The function is to provide more flexible mapping. For example if "fts" is unique and meaningful we may allow to use it for "FTSlaveProcess". I try to simplify input parameter. The real name "FTSlaveProcess" is too long.